src/leakbox.rs

```diff
@@ -33,6 +34,10 @@ impl<T> LeakBox<T> {
         self.ptr.cast()
     }
 
+    pub fn non_null(&self) -> NonNull<T> {
+        NonNull::new(self.ptr.cast::<T>()).unwrap()
```

Comment:
this is almost certainly a premature optimization, when this conversion happens in a hot path, i like to write something like
```suggestion
        #[cfg(debug_assertions)]
        NonNull::new(self.ptr.cast::<T>()).unwrap()
        #[cfg(not(debug_assertions))]
        NonNull::new_unchecked(self.ptr.cast::<T>())
```